tools/detect_python.sh

```diff
@@ -10,21 +10,30 @@
 OLDEST_PYTHON_SUPPORTED_MAJOR=3
 OLDEST_PYTHON_SUPPORTED_MINOR=7
 
-ESP_PYTHON=python
+ESP_PYTHON=not_found
 
 for p_cmd in python3 python python3.7 python3.8 python3.9 python3.10 python3.11 python3.12; do
-    $p_cmd --version >/dev/null 2>&1 || continue
+    $p_cmd --version >/dev/null 2>&1 || {
+        echo "Command \"$p_cmd\" not found" 
+        continue
+    }
     echo "Checking \"$p_cmd\" ..."
 
-    $p_cmd -c "import sys; exit(1) if sys.version_info.major < int(\"$OLDEST_PYTHON_SUPPORTED_MAJOR\") else exit(0);" || continue
-    $p_cmd -c "import sys; exit(1) if sys.version_info.minor < int(\"$OLDEST_PYTHON_SUPPORTED_MINOR\") else exit(0);" || continue
-
+    $p_cmd -c "import sys; exit(1) if sys.version_info.major < int(\"$OLDEST_PYTHON_SUPPORTED_MAJOR\") else exit(0);" || {
+        echo "\"$p_cmd\" has an unsupported major version"
+        continue
+    }
+    $p_cmd -c "import sys; exit(1) if sys.version_info.minor < int(\"$OLDEST_PYTHON_SUPPORTED_MINOR\") else exit(0);" || {
+        echo "\"$p_cmd\" has an unsupported minor version"
+        continue
+    }
+    echo "\"$p_cmd\" is a supported version"
```

Comment:
Extra print statements means output is different to `.fish` script. Also will add extra noise to log while installing - not sure if that's a good thing